sdk/communication/azure-communication-chat/azure/communication/chat/_shared/communication_identifier_serializer.py

```diff
@@ -32,12 +32,15 @@ def serialize(cls, communicationIdentifier):
         if isinstance(communicationIdentifier, PhoneNumberIdentifier):
             return CommunicationIdentifierModel(
                 kind=CommunicationIdentifierKind.PhoneNumber,
-                id=communicationIdentifier.phone_number
+                id=communicationIdentifier.identifier,
+                phone_number=communicationIdentifier.phone_number
             )
         if isinstance(communicationIdentifier, MicrosoftTeamsUserIdentifier):
             return CommunicationIdentifierModel(
                 kind=CommunicationIdentifierKind.MicrosoftTeamsUser,
-                id=communicationIdentifier.user_id
+                id=communicationIdentifier.identifier,
+                microsoft_teams_user_id=communicationIdentifier.user_id,
+                communication_cloud_environment=communicationIdentifier.cloud
             )
 
         return CommunicationIdentifierModel(
```

Comment:
The logic here should be different. If it's an instance of UnknownIdentifier, then serialize as CommunicationIdentifierKind.Unknown, else throw an exception.

https://github.com/Azure/azure-sdk-for-net/blob/8f39b465447bc25570381faa25dbafc3dd8fc3c2/sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs#L69-L73